pkgs/tools/networking/wireguard-tools/default.nix

```diff
@@ -35,18 +36,27 @@ stdenv.mkDerivation rec {
     "WITH_WGQUICK=yes"
   ];
 
-  postFixup = ''
-    substituteInPlace $out/lib/systemd/system/wg-quick@.service \
-      --replace /usr/bin $out/bin
-  '' + lib.optionalString stdenv.isLinux ''
-    for f in $out/bin/*; do
-      # Which firewall and resolvconf implementations to use should be determined by the
-      # environment, we provide the "default" ones as fallback.
-      wrapProgram $f \
-        --prefix PATH : ${lib.makeBinPath [ procps iproute2 ]} \
-        --suffix PATH : ${lib.makeBinPath [ iptables openresolv ]}
-    done
-  '';
+  postFixup =
+    let
+      # Which firewall and resolvconf implementations to use should be
+      # determined by the environment, we provide the "default" ones as
+      # fallback.
+      wrapProgramFlags = [
+        ''--suffix PATH : ${lib.makeBinPath (
+          [ wireguard-go ] ++
```

Comment:
open question: Linux?

Reply:
Definitely not on Linux, `wireguard-go` is a userspace implementation of WireGuard which isn't needed on Linux because there's a kernel module available for it (or even in Linux itself since 5.6)

Reply:
Is that a darwin only dependency?

Reply:
Not entirely sure, but it doesn't make any sense for us to ship it on Linux considering that our default kernel is 6.1 which means that the kernel module for wireguard exists in there and wg-quick only falls back to wireguard-go if the module doesn't exist.

Reply:
@Ma27  thank you! I will add a isDarwin guard

@SuperSandro2000 it is indeed a darwin dependency. Or at least: wireguard fails on my (Darwin) machine without it, with an explicit error about missing this binary.

Reply:
But to be clear, this PR basically just reverts #160951, which might not be a good idea. As it stands `wg-quick` on darwin is broken.

Reply:
It turns out that the revert button on that PR isn't clickable, so doing the equivalent without rebuilds on Linux in this PR.

Reply:
Thanks @wegank